### PR TITLE
NIFI-12572 Refactor components in azure bundle using current API methods

### PR DESCRIPTION
**Pull request by EndzeitBegins**
# Summary

Refactor tests and components in `azure` bundle to use updated APIs from [NIFI-12452](https://issues.apache.org/jira/browse/NIFI-12452).

Additionally, makes use of newer Java APIs, e.g. `List.of` and `Set.of` for declaration of relationships and properties.

[NIFI-12572](https://issues.apache.org/jira/browse/NIFI-12572)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/shared/azure/eventhubs/AzureEventHubTransportType.java [hunk 1]
```diff
@@ -16,23 +16,24 @@
...
-    AMQP("Amqp", "AMQP over TCP on ports 5671 and 5672"),
-
-    AMQP_WEB_SOCKETS("AmqpWebSockets", "AMQP over HTTPS with WebSockets on port 443");
+    AMQP("Amqp", "AMQP over TCP on ports 5671 and 5672", AmqpTransportType.AMQP),
+    AMQP_WEB_SOCKETS("AmqpWebSockets", "AMQP over HTTPS with WebSockets on port 443", AmqpTransportType.AMQP_WEB_SOCKETS);
 
     private final String value;
-
     private final String description;
+    private final AmqpTransportType azureType;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/shared/azure/eventhubs/AzureEventHubTransportType.java:31]: This should be renamed to match the type `amqpTransportType`.

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/shared/azure/eventhubs/AzureEventHubTransportType.java [hunk 2]
```diff
@@ -16,23 +16,24 @@
...
 
-    AzureEventHubTransportType(final String value, final String description) {
+    AzureEventHubTransportType(final String value, final String description, final AmqpTransportType azureType) {
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/shared/azure/eventhubs/AzureEventHubTransportType.java:33]: This enum is intentionally loosely associated to the `AmqpTransportType`, but introducing this direct connection seems useful, I would just recommend changing the property name to match the type name.

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/MetricsBuilder.java [hunk 3]
```diff
@@ -191,5 +154,20 @@ public String getTags() {
         return this.tags;
     }
 
-
+    private Metric buildMetric(String metricName) {
+        final Metric metric;
+        if(isProcessorMetric) {
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/MetricsBuilder.java:159]: Spacing:
```suggestion
        if (isProcessorMetric) {
```

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/MetricsBuilder.java [hunk 4]
```diff
@@ -191,5 +154,20 @@ public String getTags() {
...
+            metric =  new Metric(this.computer, this.processGroupId, this.processGroupName);
+            metric.setProcessorId(this.processorId);
+            metric.setProcessorName(this.processorName);
+        } else {
+            metric = new Metric(this.computer, this.processGroupId, this.processGroupName);
+        }
+        metric.setCategoryName(this.categoryName);
+        metric.setName(metricName);
+        if(this.tags != null) {
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/MetricsBuilder.java:168]:
```suggestion
        if (this.tags != null) {
```

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the cleanup and improvements @EndzeitBegins. The majority of these changes look good, I just noted a couple very minor adjustment recommendations.

**Comment by EndzeitBegins** [comment]
Thank you for your reviews of this and the other PRs @exceptionfactory. Also for taking on the merge process.

I've supplied a commit that applies your suggestions.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @EndzeitBegins, the latest version looks good! +1 merging

**Pull request closed without merging**